Added parameter to control max reionization z externally

mraveri opened the pull request:
This small change allows to set max reionization redshift from cobaya - avoiding the annoying "Did not converge to optical depth" error.

max z effectively sets an implicit prior on tau so it is useful to control it.

cmbant commented:
Thanks, though this is actually a property of BaseTauWithHeReionization not just tanh reionization?

mraveri commented:
Right, shoud I modify implementing set_extra_params for the base class and call super in the parents?

cmbant commented:
I guess something like that would better fit the structure

mraveri commented:
Implemented! Cobaya kwargs inspection prevents just passing down generic kwargs so the variable name is exposed in the child classes.

Added missing docstrings!
